--- grive-0.3.0/libgrive/src/drive/State.cc.orig	2014-09-30 12:52:11.194212473 +0100
+++ grive-0.3.0/libgrive/src/drive/State.cc	2014-09-30 12:54:24.420202658 +0100
@@ -248,12 +248,12 @@
 void State::Write( const fs::path& filename ) const
 {
 	Json last_sync ;
-	last_sync.Add( "sec",	Json(m_last_sync.Sec() ) );
-	last_sync.Add( "nsec",	Json(m_last_sync.NanoSec() ) );
+	last_sync.Add( "sec",	Json((boost::int64_t)m_last_sync.Sec() ) );
+	last_sync.Add( "nsec",	Json((boost::uint64_t)m_last_sync.NanoSec() ) );
 	
 	Json result ;
 	result.Add( "last_sync", last_sync ) ;
-	result.Add( "change_stamp", Json(m_cstamp) ) ;
+	result.Add( "change_stamp", Json((boost::int64_t)m_cstamp) ) ;
 	
 	std::ofstream fs( filename.string().c_str() ) ;
 	fs << result ;
